lib/kafo/configuration.rb

```diff
@@ -104,6 +104,10 @@ def set_custom_fact(key, value)
       custom_fact_storage[key.to_s] = value
     end
 
+    def has_custom_fact?(key)
+      custom_fact_storage.key?(key.to_sym)
```

Comment:
This is turning `key` into a symbol, if I look at (https://github.com/theforeman/kafo/pull/266/files#diff-b3325340a3c63ce60780a66f5db027d5R100) that code is ensuring the key is a string using `to_s`.

Reply:
Thanks @ehelms . Updated and also added a first pass at a test case for the hook context.